src/bluesearch/entrypoint/database/add.py

```diff
@@ -62,6 +65,50 @@ def run(
     with open(path, "rb") as f:
         article = pickle.load(f)  # nosec
 
+    # Article table.
+
+    # TODO At the moment, no identifiers are extracted. This is a patch waiting for it.
+    article_id = generate_uid((article.title,))
+
+    article_mapping = {
+        "article_id": article_id,
+        "title": article.title,
+        "authors": ", ".join(article.authors),
+        "abstract": "\n".join(article.abstract),
+    }
+    article_keys = article_mapping.keys()
+    article_fields = ", ".join(article_keys)
+    article_binds = f":{', :'.join(article_keys)}"
+
+    with engine.connect() as con:
+        query = sqlalchemy.text(
+            f"INSERT INTO articles({article_fields}) VALUES({article_binds})"
+        )
+        con.execute(query, article_mapping)
+
+    # Sentence table.
+
+    sentence_mappings = []
+    swapped = ((text, section) for section, text in article.section_paragraphs)
+    nlp = load_spacy_model("en_core_sci_lg", disable=["tagger", "ner"])
+    for pposition, (document, section) in enumerate(nlp.pipe(swapped, as_tuples=True)):
+        for sposition, sentence in enumerate(document.sents):
+            sentence_mapping = {
+                "section_name": section,
+                "text": sentence.text,
+                "article_id": article_id,
+                "paragraph_pos_in_article": pposition,
+                "sentence_pos_in_paragraph": sposition,
+            }
+        sentence_mappings.append(sentence_mapping)
+
+    sentences_keys = sentence_mapping.keys()
```

Comment:
`sentence_mapping` might not be defined at this point (if the for-loop is zero length)

Reply:
Solution in 5787623.